tests/test_utils.py

```diff
@@ -281,3 +281,224 @@ def dir_name_to_display_name(dir_name: str):
+        ),
+        (
+            "All scenarios tagged with exclude cause no included scenarios",
+            Feature([], scenarios_tagged_with_a),
+            None,
+            ["a"],
+            [],
+        ),
+        (
+            "All scenarios tagged with both include and exclude "
+            "cause no included scenarios",
+            Feature([], scenarios_tagged_with_a_and_b),
+            ["a"],
+            ["b"],
+            [],
+        ),
+        (
+            "Only scenarios tagged with an include tag are included",
+            Feature([], scenarios_tagged_with_a + scenarios_tagged_with_b),
+            ["a"],
+            None,
+            scenarios_tagged_with_a,
+        ),
+        (
+            "Only scenarios tagged with an exclude tag are excluded",
+            Feature([], scenarios_tagged_with_a + scenarios_tagged_with_b),
+            None,
+            ["a"],
+            scenarios_tagged_with_b,
+        ),
+        # ~ FEATURE & SCENARIO TAGS ~
+        (
+            "When a feature is tagged with an include tag, "
+            "only scenarios tagged with an exclude tag are excluded",
+            Feature(["a"], scenarios_tagged_with_a + scenarios_tagged_with_b),
+            None,
+            ["b"],
+            scenarios_tagged_with_a,
+        ),
+        (
+            "When a feature is tagged with exclude tag, "
+            "but scenarios have include tag, "
+            "then no scenarios are included",
+            Feature(["b"], scenarios_tagged_with_a + scenarios_tagged_with_b),
+            ["a"],
+            ["b"],
+            [],
+        ),
+        # ~ EXAMPLES TABLES TAGS ~
+        (
+            "Specific examples tables can be excluded",
+            Feature(
+                tags=["a"],
+                scenarios=[
+                    Scenario(
+                        tags=[],
+                        examples=examples_tagged_with_a + examples_tagged_with_b,
+                    )
+                ],
+            ),
+            ["a"],
+            ["b"],
+            [Scenario(tags=[], examples=examples_tagged_with_a)],
+        ),
+        (
+            "Scenarios with include tag, but with all examples tables "
+            "having an exclude tag will exclude the whole scenario",
+            Feature(
+                tags=["a"],
+                scenarios=[Scenario(tags=[], examples=examples_tagged_with_b)],
+            ),
+            ["a"],
+            ["b"],
+            [],
+        ),
+        (
+            "Scenario is included if only an examples table has an include tag",
+            Feature(
+                tags=[],
+                scenarios=[
+                    Scenario(
+                        tags=[],
+                        examples=examples_tagged_with_a + examples_tagged_with_b,
+                    )
+                ],
+            ),
+            ["a"],
+            ["b"],
+            [Scenario(tags=[], examples=examples_tagged_with_a)],
```

Comment:
I love the table driven nature of this, but frankly there was a lot of parallelism going on and i just gave up trying to make sure all the hard coded tags and the names containing the tag name(s) were consistent.

Reply:
Honestly I felt like it might be useful to have these tests be written in Gherkin, but somewhat ironically that isn't set up in this repo. The gherkin would do a nice job of pulling out the non-essential/arbitrary values and focus on the difference in variations. The `description` is meant to be the scenario names essentially, so one could use those as the guide for determining if there are any variations missing.